### DISCUSSION TITLE
Google analytics problem

### DISCUSSION THREAD

**Discussion by nibinglobal** [breezefront/community#22] [1. Help]
Google Analytics is not working properly if we enable SPA. We do not get details of the categories and product pages that customers use.
is it normal?
Also, Sometimes magento spins forever even when fully loaded.

**Comment by vovayatsyuk** [comment]
Did you check the network panel to see if GoogleAnalytics requests are being sent when browsing the store?

**Reply by nibinglobal** [reply]
It works well on non-spa pages like home page,cart etc.
Also, check the screen shot below.
Spinning error
![Screenshot 2022-10-10 at 6 18 08 PM](https://user-images.githubusercontent.com/78966425/194870235-9f999d0c-8049-48ce-905d-76a10c0e4b18.png)

**Reply by vovayatsyuk** [reply]
Maybe you have some js errors? Please share the link to your site.

**Reply by nibinglobal** [reply]
https://www.brihatha.in

**Reply by vovayatsyuk** [reply]
I don't see the spinner. How can I reproduce the issue?

**Comment by nibinglobal** [comment]
Can you click on different categories?

**Reply by vovayatsyuk** [reply]
I clicked on about 10 different categories, cart, account, and a few more categories again.

**Comment by nibinglobal** [comment]
Thank you for your quick reply
I don't know why it doesn't happen now. I will let you know if this happens again. I tested it with Firefox and Safari browsers.
What about Google Analytics?
Can you help me to overcome the problem of google analytics?

**Reply by vovayatsyuk** [reply]
Looks like you are using manual ga integration (js code inside miscellaneous option).

Depending on which version of GA you use you may solve it in different ways: https://www.analyticsmania.com/post/single-page-web-app-with-google-tag-manager/

**Comment by nibinglobal** [comment]
i am using weltpixel enhaced analytics

**Reply by vovayatsyuk** [reply]
This doesn't change the way to solve the issue. Have you read the article? Especially "Method 1. Track pageviews with GA4 Enhanced Measurement"

**Reply by nibinglobal** [reply]
I had read it

**Comment by vovayatsyuk** [comment]
For those who are trying to setup GA with turbo. GA4 is compatible with turbo out of the box since it tracks history change events by default.

If for some reason it doesn't work for you, please check the following GA setting:

![CleanShot 2022-10-26 at 18 17 15@2x](https://user-images.githubusercontent.com/306080/198067126-e7f7d424-c8ab-4b31-a610-8a3470d920df.png)
![CleanShot 2022-10-26 at 18 19 04@2x](https://user-images.githubusercontent.com/306080/198067138-e19f3423-7a2c-40ae-9def-49b324148573.png)
![CleanShot 2022-10-26 at 18 19 54@2x](https://user-images.githubusercontent.com/306080/198067141-8825b81e-e483-48ca-9951-87393101cd3a.png)

**Reply by ShubWCL** [reply]
Thank you for this but the settings are already enabled but the tracking still not working in GA4 too.
